src/Process/Signal.php

```diff
@@ -100,10 +110,35 @@ public function handleSignal(int $signalNumber, $signalInformation): void
                 // For now, SITGTERM is safe.
                 $this->processSignalInformation($information);
             } else {
-                $this->bufferedSignals[] = $information;
+                $this->bufferSignalInformation($information);
             }
         }
 
         return;
     }
+
+    protected function bufferSignalInformation(InformationInterface $information): SignalInterface
+    {
+        if ($this->getCanBufferSignals()) {
+            $this->bufferedSignals[] = $information;
+        }
+
+        return $this;
+    }
+
+    public function setCanBufferSignals(bool $canBufferSignals): SignalInterface
+    {
+        if ($this->canBufferSignals !== $canBufferSignals) {
```

Comment:
**blocker**

You initialized `$this->canBufferSignals` to `true` and this logic prevents it from being set to `false`

For cases where I only want a value to be set once, but fall back to a default, I like

```
protected const DEFAULT_VALUE = <whatever>;

protected $value;

protected function getValue()
{
    if ($this->value === null) {
        $this->value = static::DEFAULT_VALUE;
    }
    return $this->value;
}

public function setValue($value) {
    if ($this->value !== null) {
        throw \LogicException();
    }
    $this->value = $value;
    return $this;
}
```

We discussed it recently in LS [here](https://github.com/neighborhoods/ListingService/pull/361#discussion_r248836424)

Reply:
No. This allows a user to set it to false if it is true, or if it is false back to true but not false again and the same for the inverse.

Reply:
What's the point, then? It's an idempotent interaction, so I don't really see a reason to block a user from being explicit that they want it to be a certain way, regardless of what the default is. Otherwise changing the default creates a breaking change that the user can't even prepare for in advance.

Reply:
> What's the point, then? 

I'd like to know if I'm setting it again, but this can be turned on and off. By default it is on.

> Otherwise changing the default creates a breaking change that the user can't even prepare for in advance.

I'm sorry, I don't understand this sentence.

At the end of the day I don't really care as long as this can be toggled. I thought it nice for Signal to tell me that I was doing something twice since that shouldn't be my intention. The getter is exposed, so US should have all the tools that they need to not reset the same value.

Reply:
The BC is that if it's set to true by default and something wants it set to false, they just call
```
$signal->setCanBufferSignals(false);
```
But, if the default changes to false, their code suddenly breaks unless everywhere they use a setter, they first have to use the getter, which feels like an antipattern for simple, idempotent processes. Most things don't care what state something was previously in, they just care about making that thing be in the state they desire.

Reply:
sweet, I will remove the logical assertion